lib/utils/style-gather.html

```diff
@@ -123,7 +123,9 @@
           // other dom-modules that contain styling
           let include = e.getAttribute(INCLUDE_ATTR);
           if (include) {
-            styles.push(...new Set(this.stylesFromModules(include)));
+            styles.push(...this.stylesFromModules(include).filter(function(item, index, self) {
```

Comment:
This looks O(n<sup>2</sup>) to me
I think this would be faster as
```js
let stylesFromModules = this.styleFromModules(include);
let uniqueStyles = Object.keys(this.stylesFromModules(include).reduce((acc, s) => {acc[s] = true; return acc}, {}));
styles.push(...uniqueStyles);
```

Reply:
oh this wouldn't work because the array is of `<style>` elements.

Reply:
Since most arrays will be small here, I think we can keep this for now. We can always revisit later (or just use Set someday...)